qiskit/transpiler/preset_passmanagers/level0.py

```diff
@@ -18,43 +18,23 @@
 from qiskit.transpiler.passmanager_config import PassManagerConfig
 from qiskit.transpiler.timing_constraints import TimingConstraints
 from qiskit.transpiler.passmanager import PassManager
+from qiskit.transpiler.passmanager import StructuredPassManager
 
-from qiskit.transpiler.passes import Unroller
-from qiskit.transpiler.passes import BasisTranslator
-from qiskit.transpiler.passes import UnrollCustomDefinitions
-from qiskit.transpiler.passes import Unroll3qOrMore
-from qiskit.transpiler.passes import CheckMap
-from qiskit.transpiler.passes import GateDirection
 from qiskit.transpiler.passes import SetLayout
 from qiskit.transpiler.passes import TrivialLayout
 from qiskit.transpiler.passes import DenseLayout
 from qiskit.transpiler.passes import NoiseAdaptiveLayout
 from qiskit.transpiler.passes import SabreLayout
-from qiskit.transpiler.passes import BarrierBeforeFinalMeasurements
 from qiskit.transpiler.passes import BasicSwap
 from qiskit.transpiler.passes import LookaheadSwap
 from qiskit.transpiler.passes import StochasticSwap
 from qiskit.transpiler.passes import SabreSwap
-from qiskit.transpiler.passes import FullAncillaAllocation
-from qiskit.transpiler.passes import EnlargeWithAncilla
-from qiskit.transpiler.passes import ApplyLayout
-from qiskit.transpiler.passes import CheckGateDirection
-from qiskit.transpiler.passes import Collect2qBlocks
-from qiskit.transpiler.passes import ConsolidateBlocks
-from qiskit.transpiler.passes import UnitarySynthesis
-from qiskit.transpiler.passes import TimeUnitConversion
-from qiskit.transpiler.passes import ALAPSchedule
-from qiskit.transpiler.passes import ASAPSchedule
-from qiskit.transpiler.passes import AlignMeasures
-from qiskit.transpiler.passes import ValidatePulseGates
-from qiskit.transpiler.passes import PulseGates
 from qiskit.transpiler.passes import Error
-from qiskit.transpiler.passes import ContainsInstruction
-
+from qiskit.transpiler.preset_passmanagers import common
 from qiskit.transpiler import TranspilerError
 
 
-def level_0_pass_manager(pass_manager_config: PassManagerConfig) -> PassManager:
+def level_0_pass_manager(pass_manager_config: PassManagerConfig) -> StructuredPassManager:
```

Comment:
If you have your structured pass, is there any need for these not very user friendly functions which take a pass manager config?

Could you instead just have `default_pass_manager(**all_the_transpile_kwargs) -> StructuresPassManager`
that can be used like you would use the transpile function but instead returns the configured pass manager for those kwarg values. 

As they are these functions are basically unusable by an end user due to the PassManagerConfig. Building that object (which is currently done by transpile) could be instead done here. And then transpile/_transpile_circuit could just be a thin wrapper for these default generators

Reply:
I think we can move towards that in a follow up, restructuring this whole path is out of scope for this. I originally just had this PR as adding the new structure pass manager class and save doing just this restructuring for a follow up but was asked during the initial review to use it for at least this. I don't want to increase the scope for the PR more than this here.

Part of the issue here is also around how terrible the parallel dispatch and arg handling is in `transpile()`, I'm currently fighting that in #7789. I don't think this PR is the complete story around how we create composable, structured, pluggable pass managers. It's just the first step.

Reply:
I did this in: https://github.com/Qiskit/qiskit-terra/pull/8125 because it's really a standalone thing that doesn't depend on this. The only change this PR causes is changing the return type of that function from a `PassManager` -> `StagedPassManager`.